src/QMCWaveFunctions/EinsplineSetBuilderESHDF.fft.cpp

```diff
@@ -124,8 +126,23 @@ bool EinsplineSetBuilder::ReadOrbitalInfo_ESHDF(bool skipChecks)
   // Read atom orbital info from xml //
   /////////////////////////////////////
   // construct Super2Prim mapping.
-  if (Super2Prim.size() == 0)
-  {
+  // Check the number of ions in the XML and 
+  // HDF5. Skip the checks in some situations but 
+  // not others. See Issue #2224.
+
+  if (SourcePtcl->R.size() != IonPos.size()) {
+    app_error() << "The ion sizes differ between the XML and HDF5.\n" 
+                << "Number of ions in XML : " << SourcePtcl->R.size() << std::endl
+                << "Number of ions in HDF5: " << IonPos.size() << std::endl;
+    abort();
+  }
```

Comment:
This check should be removed.    The QMC wavefunction can be created by tiling the DFT wavefunction, in which case the number of ions will not match.    I suppose this could be checked with (# of tiles)*(DFT Ions) == (QMC ions), but that's not necessary now.

Reply:
If I remove this check, the wavefunction unit tests pass for my build.

Reply:
I'm confused about how to proceed, as mentioned above, I don't even get to this error now. I hit another error.

Reply:
With the new change made by @ye-luo, do I need to define an empty particle set?